anyway/parsers/location_extraction.py

```diff
@@ -134,43 +133,57 @@ def geocode_extract(location):
     district = None
     address = None
     geom = {"lat": None, "lng": None}
-    try:
-        gmaps = googlemaps.Client(key=secrets.get("GOOGLE_MAPS_KEY"))
-        geocode_result = gmaps.geocode(location, region="il")
-        if geocode_result is None or geocode_result == []:
-            return None
-        response = geocode_result[0]
-        geom = response["geometry"]["location"]
-        for item in response["address_components"]:
-            if "route" in item["types"]:
-                if item["short_name"].isdigit():
-                    road_no = int(item["short_name"])
-                else:
-                    street = item["long_name"]
-            elif "point_of_interest" in item["types"] or "intersection" in item["types"]:
-                intersection = item["long_name"]
-            elif "locality" in item["types"]:
-                city = item["long_name"]
-            elif "administrative_area_level_2" in item["types"]:
-                subdistrict = item["long_name"]
-            elif "administrative_area_level_1" in item["types"]:
-                district = item["long_name"]
-        address = response["formatted_address"]
-        if road_no is None and extract_road_number(location) is not None:
-            road_no = extract_road_number(location)
-    except Exception as _:
-        logging.exception("geocode extract location {}".format(location))
+    location_gen = get_rectified_location_strings(location)
+    while location:
```

Comment:
Why not `for location in get_rectified_location_strings(location)`? Any decision (such as not yielding any strings) can be moved inside the generator. Also, I would use the name `candidate_location_strings()`.

Reply:
because I didn't proof read 🤦 
Fixed of course